cpp/include/cuvs/preprocessing/quantize/pq.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cuvs/neighbors/common.hpp>
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/handle.hpp>
+#include <raft/core/host_mdspan.hpp>
+
+namespace cuvs::preprocessing::quantize::pq {
+
+/**
+ * @defgroup pq Product Quantizer utilities
+ * @{
+ */
+
+/**
+ * @brief Product Quantizer parameters.
+ * @see cuvs::neighbors::vpq_params
+ */
+using params = cuvs::neighbors::vpq_params;
+
+/**
+ * @brief Defines and stores VPQ codebooks upon training
+ *
+ * @tparam T data element type
+ *
+ */
+template <typename T>
+struct quantizer {
+  params params_quantizer;
+  cuvs::neighbors::vpq_dataset<T, int64_t> vpq_codebooks;
+};
+
+/**
+ * @brief Initializes a product quantizer to be used later for quantizing the dataset.
+ *
+ * Usage example:
+ * @code{.cpp}
+ * raft::handle_t handle;
+ * cuvs::preprocessing::quantize::pq::params params;
```

Comment:
According to our offline discussion, here we might want to update the example to show how a pool allocator can be used to improve training time.